src/sensor_poll.py

```diff
@@ -80,6 +82,10 @@ def __init__(self, katcp_ip="10.103.254.6", katcp_port=7147):
             Port to connect to! [Defaults: 7147]
         """
         self.katcp_ip = katcp_ip
+        try:
+            self.hostname = ''.join(socket.gethostbyaddr(katcp_ip)[1])
+        except Exception:
+            self.hostname = 'Unknown'
```

Comment:
If `hostname == 'Unknown'` do you still want to invoke `_kcp_connect`?

Reply:
Yes, with an unknown hostname.